rs/index/src/posting_list/file.rs

```diff
@@ -0,0 +1,699 @@
+                current_file_num += 1;
+                current_offset = 0;
+                if current_file_num >= self.mmaps.len() {
+                    return Err(anyhow!("Current file nunber out of bound"));
+                }
+            }
+        }
+
+        Ok(posting_list)
+    }
+
+    fn append(&mut self, posting_list: &[u64]) -> Result<()> {
+        if self.current_num_of_posting_list == self.num_clusters {
+            return Err(anyhow!(
+                "Trying to append more posting lists than number of clusters"
+            ));
+        }
+        let required_size = posting_list.len() * std::mem::size_of::<u64>();
+        self.size_bytes += required_size;
+        let should_flush = self.resident && self.size_bytes > self.memory_threshold;
+        let flush = should_flush && !self.resident_posting_lists.is_empty();
+
+        // Good case, where file is still resident
+        if self.resident && !should_flush {
+            self.current_num_of_posting_list += 1;
+            self.resident_posting_lists.push(posting_list.to_vec());
+            return Ok(());
+        }
+
+        // Spill to disk, creating new files if necessary
+        let pl_metadata_in_bytes = 2 * std::mem::size_of::<u64>();
+        if flush {
+            self.flush_resident_posting_lists_to_disk()?;
+            // At this point we are not in memory anymore, we'll need to
+            // take into account the storage for metadata.
+            self.size_bytes += self.resident_posting_lists.len() * pl_metadata_in_bytes;
+        }
+
+        self.size_bytes += pl_metadata_in_bytes;
+        // We should spill to disk, but did not flush (there was nothing to flush)
+        if self.resident && !flush {
+            // Trigger the creation of a new file since this is the first time we write
+            // to disk
+            self.current_offset = self.backing_file_size;
+            // Flip the flag since we are technically on disk at this point
+            self.resident = false;
+        } else {
+            // Adjust the offset to write metadata first
+            self.current_offset = self.current_num_of_posting_list * pl_metadata_in_bytes;
+        }
+        self.append_posting_list_to_disk(posting_list)?;
+        Ok(())
+    }
+
+    fn write(&mut self, writer: &mut std::io::BufWriter<&mut std::fs::File>) -> Result<usize> {
+        let mut total_bytes_written = writer.write(&self.len().to_le_bytes())?;
+
+        // If the data is still resident in memory, flush it to disk first
+        if self.resident {
+            self.flush_resident_posting_lists_to_disk()?;
+        }
+
+        for (i, mmap) in self.mmaps.iter().enumerate() {
+            let bytes_to_write = if i as i32 == self.current_backing_id {
+                self.current_offset
+            } else {
+                mmap.len()
+            };
+
+            let bytes_written = writer.write(&mmap[..bytes_to_write])?;
+            total_bytes_written += bytes_written;
```

Comment:
```suggestion
            total_bytes_written += wrap_write(writer, ...)?;
```

Reply:
`bytes_written` is actually used for detecting error further down